packages/pigeon/platform_tests/shared_test_plugin_code/lib/integration_tests.dart

```diff
@@ -488,6 +488,18 @@ void runPigeonIntegrationTests(TargetGenerator targetGenerator) {
       final String echoObject = await api.echoAsyncString(sentObject);
       expect(echoObject, sentObject);
     });
+
+    testWidgets('async errors are returned correctly', (WidgetTester _) async {
```

Comment:
Please make this two tests; the fewer things a single unit test tests, the better.